skfem/io/meshio.py

```diff
@@ -114,18 +117,47 @@ def from_meshio(m,
 
     # parse boundaries from cell_sets
     if m.cell_sets and bnd_type in m.cells_dict:
-        facets = {
-            k: [tuple(f) for f in np.sort(m.cells_dict[bnd_type][v[bnd_type]])]
+        oriented_facets = {
+            k: [tuple(f) for f in m.cells_dict[bnd_type][v[bnd_type]]]
             for k, v in m.cell_sets_dict.items()
             if bnd_type in v and k.split(":")[0] != "gmsh"
         }
-        boundaries = {k: np.array([i for i, f in
-                                   enumerate(map(tuple, mtmp.facets.T))
-                                   if f in v])
-                      for k, v in facets.items()}
+        sorted_facets = {k: [tuple(np.sort(f)) for f in v]
+                         for k, v in oriented_facets.items()}
+
+        for k, v in oriented_facets.items():
+            indices = []
+            oris = []
+            for i, f in enumerate(map(tuple, mtmp.facets.T)):
+                if f in sorted_facets[k]:
+                    indices.append(i)
+                    ix = sorted_facets[k].index(f)
+                    facet = v[ix]
+                    t1, t2 = mtmp.f2t[:, i]
+                    if t2 == -1:
+                        # orientation on boundary is 0
+                        oris.append(0)
+                        continue
+                    if len(f) == 2:
+                        # rotate tangent to find normal
+                        tangent = mtmp.p[:, facet[1]] - mtmp.p[:, facet[0]]
+                        normal = np.array([-tangent[1], tangent[0]])
+                    elif len(f) == 3:
+                        # cross product to find normal
+                        tangent1 = mtmp.p[:, facet[1]] - mtmp.p[:, facet[0]]
+                        tangent2 = mtmp.p[:, facet[2]] - mtmp.p[:, facet[0]]
+                        normal = -np.cross(tangent1, tangent2)
+                    else:
```

Comment:
Here is the part reading orientation from Gmsh.

Reply:
Comparing our normal vs Gmsh normal is done at https://github.com/kinnala/scikit-fem/pull/865/files#diff-b63e72d0a404f77693c095a4cd86504fe38dcc73d6c9dc4cd1d8f9d4157c0463R156